src/main/java/com/google/sps/servlets/EventServlet.java

```diff
@@ -75,6 +89,116 @@ public void doPut(HttpServletRequest request, HttpServletResponse response) thro
     response.sendRedirect("/index.html");
   }
 
+  private void addEvent(HttpServletRequest request, HttpServletResponse response, UserService userService)
+      throws IOException, ServletException {
+    String title = Objects.toString(request.getParameter("title"), "");
+    String description = Objects.toString(request.getParameter("description"), "");
+    String duration_parameter = request.getParameter("duration");
+    Long duration = 0L;
+    if (duration_parameter != null && !duration_parameter.isEmpty()) {
+      try {
+        duration = Long.parseLong(duration_parameter);
+      } catch (Exception e) {
+        System.err.println(e + duration_parameter);
+      }
+    }
+
+    String location = Objects.toString(request.getParameter("location"), "");
+
+    Map<String, String> fields = new HashMap<String, String>();
+    if (request.getParameterValues("fields") != null) {
+      for (String field : request.getParameterValues("fields")) {
+        fields.put(field, request.getParameter(field));
+      }
+    }
+
+    List<String> tags = new ArrayList<String>();
+    if (request.getParameterValues("tags") != null) {
+      tags = Arrays.asList(request.getParameterValues("tags"));
+    }
+
+    String current_user_id = userService.getCurrentUser().getUserId();
+    List<String> participants_ids = new ArrayList<String>();
+    if (request.getParameterValues("people") != null) {
+      participants_ids = Arrays.asList(request.getParameterValues("people"))
+                               .stream().map(person -> UserStorage.getIDbyUsername(person)).collect(Collectors.toList());
+    }
+
+    String startDateAsString = request.getParameter("start-date");
+    String startTimeAsString = request.getParameter("start-time");
```

Comment:
I think using a string here makes it prone to errors because this call depends on the formatting of this string.
I would suggest to accept only unix epoch in seconds as input. Then you also need only a single param and not two.